lms/djangoapps/course_api/blocks/tests/test_api.py

```diff
@@ -77,3 +75,17 @@ def test_sub_structure(self):
                 self.assertIn(unicode(block.location), blocks['blocks'])
             else:
                 self.assertNotIn(unicode(block.location), blocks['blocks'])
+
+    def test_filtering_by_block_types(self):
+        sequential_block = self.store.get_item(self.course.id.make_usage_key('sequential', 'sequential_y1'))
+
+        block_types = ['problem']
+        blocks = get_blocks(self.request, sequential_block.location, self.user, block_types=block_types)
+
+        for block_type, block_name in (
+                ('problem', 'problem_y1a_1'),
+                ('problem', 'problem_y1a_2'),
+                ('problem', 'problem_y1a_3'),
+        ):
+            block = self.store.get_item(self.course.id.make_usage_key(block_type, block_name))
+            self.assertIn(unicode(block.location), blocks['blocks'])
```

Comment:
Agree.  We want to test that non-problem blocks are NOT in the returned list, while problem blocks are.

If you move this test method to `test_serializers.py`, there you can test that it works as expected for both serializers by adding a common helper in `TestBlockSerializerBase` that is called by methods in `TestBlockSerializer` and `TestBlockDictSerializer`.  You can do so by looping through all the blocks in `self.block_structure` and making sure that only blocks of type `problem` are returned by the serializer.